packages/patternfly-4/react-core/src/components/Dropdown/DropdownToggleCheckbox.tsx

```diff
@@ -28,26 +30,29 @@ export class DropdownToggleCheckbox extends React.Component<DropdownToggleCheckb
     className: '',
     isValid: true,
     isDisabled: false,
-    isChecked: null as boolean | null,
-    checked: null as boolean | null,
     onChange: () => undefined as any
   };
 
-  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.props.onChange(event.currentTarget.checked, event);
+  handleChange = (checked: boolean, event: React.FormEvent<HTMLInputElement>) => {
+    this.props.onChange(checked, event);
+  }
+
+  calculateChecked = () => {
+    const { isChecked, checked } = this.props;
+    return isChecked !== undefined ? isChecked : checked;
   }
 
   render() {
-    const { className, onChange, isValid, isDisabled, isChecked, checked, ...props } = this.props;
+    const { className, onChange, isValid, isDisabled, isChecked, ref, checked, ...props } = this.props;
     return (
       <label className={css(styles.dropdownToggleCheck, className)} htmlFor={props.id}>
-        <input
+        <Checkbox
           {...props}
-          type="checkbox"
-          onChange={this.handleChange}
+          {...(this.calculateChecked() !== undefined) && { onChange: this.handleChange }}
+          ref={ref as any}
```

Comment:
Anybody knows something better to use here?

Reply:
Why do you have to pass the ref here? It will be passed automatically with `{...props}` with the correct type.

Reply:
It's because TS does not recognized the ref correctly in here. I tried casting it to expected and it failed. This was the only solution that didn't thre any error.